File: Domain/Team.cs
namespace Domain
{
    public class Team
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public int ScoredGoals { get; set; } = 0;
        public int ConcededGoals { get; set; } = 0;
        public int Points{get;set;} = 0;
        public ICollection<Player> Players {get;set;} = new List<Player>();
    }
}

File: client-app/src/app/models/team.ts
import { Player } from "./player";

export interface ITeam {
  id: string;
  name: string;
  players?: Player[];
  scoredGoals?: number;
  concededGoals?: number;
  points?:number;
}

export class Team implements ITeam {
  constructor(init: TeamFormValues) {
    this.id = init.id!
    this.name = init.name
  }

  id: string;
  name: string;
  players?: Player[];
  bestGoalScorer?: Player
  bestAssistant?: Player
  scoredGoals?: number;
  concededGoals?: number;
  points?:number;
}

export class TeamFormValues {
  id?: string = undefined;
  name: string = '';

  constructor(team?: TeamFormValues) {
    if (team) {
      this.id = team.id;
      this.name = team.name;
    }
  }
}